cmd/installer/cli/install_config.go

```diff
@@ -176,11 +207,11 @@ func processTLSConfig(flags *installFlags, installCfg *installConfig) error {
 			return fmt.Errorf("failed to parse TLS certificate: %w", err)
 		}
 
-		installCfg.tlsCert = cert
+		installCfg.tlsCert = &cert
 		installCfg.tlsCertBytes = certBytes
 		installCfg.tlsKeyBytes = keyBytes
-	} else if installCfg.enableManagerExperience {
-		// For manager experience, generate self-signed certificate if none provided
+	} else if !flags.headless && installCfg.enableManagerExperience {
+		// For UI based manager experience, generate self-signed cert if none provided, with user confirmation
```

Comment:
don't we still need to generate certs even in headless mode because we store those in a secret and use them on upgrades or whenever the user wants to access the UI later?

Reply:
I wasn't aware of that requirement. Could we not generate them on upgrade or support headless upgrades as well?

Reply:
i think we still need certs for a headless install because if we have a `console` command or something like that that lets someone see their deployment history and view logs and stuff, then they'd need certs to access that securely. feels like it makes sense to generate on install always.

Reply:
and that's why i would suggest to simply use the tcp listener instead of a unix socket because we'll also need the port to be open anyways for visiting the console later.